src/components/TodoItem.tsx

```diff
@@ -0,0 +1,187 @@
+import cn from 'classnames';
+import { Todo } from '../types/Todo';
+import React, { useEffect, useRef, useState } from 'react';
+import { ErrorType } from '../types/ErrorType';
+
+interface TodoItemProps {
+  todo: Todo;
+  onDeleteTodo: (todoId: Todo['id']) => void;
+  isLoading: boolean;
+  onToggleStatus: (todoId: number) => void;
+  onUpdateTodo: (
+    todoId: Todo['id'],
+    data: Partial<Omit<Todo, 'id'>>,
+  ) => Promise<void>;
+  setErrorMessage: (error: ErrorType) => void;
+  isEditing: boolean;
+  onStartEditing: (todoId: number) => void;
+  onEndEditing: (todoId: number) => void;
+}
+
+export const TodoItem = ({
+  todo,
+  onDeleteTodo,
+  isLoading,
+  onToggleStatus,
+  onUpdateTodo,
+  setErrorMessage,
+  isEditing,
+  onStartEditing,
+  onEndEditing,
+}: TodoItemProps) => {
+  const [editedTitle, setEditedTitle] = useState(todo.title);
+  const [isProcessing, setIsProcessing] = useState(false);
+  const [isUpdatedError, setIsUpdatedError] = useState(false);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (isEditing) {
+      setEditedTitle(todo.title);
```

Comment:
isProcessing isEditing i'm confused, we shoudn't set state in useEffect
can we simplify
KISS